contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -201,7 +202,33 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
 		return false;
 	}
 	else
-	{	
+	{
+		/* Do not check for duplicate function if REPLACE flag is set and during dump and restore. */
+		if (!stmt->replace && !babelfish_dump_restore)
+		{
+			ObjectWithArgs *func = NULL;
+			Oid func_oid = InvalidOid;
+			char *schemaname = NULL;
+			char *funcname = NULL;
+
+			func = makeNode(ObjectWithArgs);
+			func->objname = stmt->funcname;
+			func->args_unspecified = true;
+
+			/* Get the schema name and function name. */
+			DeconstructQualifiedName(stmt->funcname, &schemaname, &funcname);
+
+			/* function, procedure */
+			func_oid = LookupFuncWithArgs(OBJECT_ROUTINE, func, true);
+
+			if (OidIsValid(func_oid))
+			{
+				/* Restrict duplicate procedure/function. */
+				ereport(ERROR, (errcode(ERRCODE_DUPLICATE_OBJECT),
```

Comment:
We should use this error "function "%s" already exists with same name". Also map it to error 2714 on SQL server side